types/stampit/index.d.ts

```diff
@@ -1,518 +1,560 @@
-
-        /**
-         * Deeply merge properties of Stamp arbitrary metadata and add them to a new
-         * Stamp and any future Stamp it composes. Creates and returns a new Stamp.
-         * Chainable.
-         * @param deepConfs The object(s) containing metadata properties
-         * @returns A new Stamp
-         */
-        deepConf(...deepConfs: Array<{}>): Stamp;
-
-        /**
-         * Deeply merge properties of Stamp arbitrary metadata and add them to a new
-         * Stamp and any future Stamp it composes. Creates and returns a new Stamp.
-         * Chainable.
-         * @param deepConfs The object(s) containing metadata properties
-         * @returns A new Stamp
-         */
-        deepConfiguration(...deepConfs: Array<{}>): Stamp;
+    /**
+     * A function used as `.initializers` argument.
+     * @template Obj The type of the object instance being produced by the `Stamp`.
+     * @template S̤t̤a̤m̤p̤ The type of the `Stamp` producing the instance.
+     */
+    interface Initializer<Obj, S̤t̤a̤m̤p̤ extends StampSignature> {
+        // ??? should the type of `this` be the object or the stamp
+        (/* this: S̤t̤a̤m̤p̤, */ options: /*_propertyMap*/ any, context: InitializerContext<Obj, S̤t̤a̤m̤p̤>): void | Obj;
+    }
 
-        /**
-         * Apply ES5 property descriptors to object instances created by the new
-         * Stamp returned by the function and any future Stamp it composes. Creates
-         * and returns a new stamp. Chainable.
-         * @param descriptors
-         * @returns A new Stamp
-         */
-        propertyDescriptors(...descriptors: Array<{}>): Stamp;
+    /**
+     * The `Initializer` function context.
+     * @template Obj The type of the object instance being produced by the `Stamp`.
+     * @template S̤t̤a̤m̤p̤ The type of the `Stamp` producing the instance.
+     */
+    interface InitializerContext<Obj, S̤t̤a̤m̤p̤ extends StampSignature> {
+        /** The object instance being produced by the `Stamp`. If the initializer returns a value other than `undefined`, it replaces the instance. */
+        instance: Obj;
+        /** A reference to the `Stamp` producing the instance. */
+        stamp: S̤t̤a̤m̤p̤;
+        /** An array of the arguments passed into the `Stamp`, including the options argument. */
+        // ! above description from the specification is obscure
+        args: any[];
+    }
 
-        /**
-         * Apply ES5 property descriptors to a Stamp and any future Stamp it
-         * composes. Creates and returns a new stamp. Chainable.
-         * @param descriptors
-         * @returns A new Stamp
-         */
-        staticPropertyDescriptors(...descriptors: Array<{}>): Stamp;
+    /**
+     * A function used as `.composers` argument.
+     * @template S̤t̤a̤m̤p̤ The type of the `Stamp` produced by the `.compose()` method.
+     */
+    interface Composer<S̤t̤a̤m̤p̤ extends StampSignature> {
+        // ??? should the type of `this` be the object or the stamp
```

Comment:
Answering the question. The `compoers` functions do not have any `this`.  Although, the specification does not mention it. https://github.com/stampit-org/stamp-specification/blob/master/README.md#composer-parameters

Reply:
Also, the return type of a composer should be `Stamp | undefined`

Reply:
`undefined`? Because it can be anything?
Or should it be `void | S̤t̤a̤m̤p̤` similar to Initializer?